lib/scripts.js

```diff
@@ -78,7 +78,8 @@ var scripts = {
     })
 
     var script = [
-      'local jobId = ARGV[2] == "" and redis.call("INCR", KEYS[5]) or ARGV[2]',
+      'local numericJobId = redis.call("INCR", KEYS[5])',
+      'local jobId = ARGV[2] == "" and numericJobId or ARGV[2]',
```

Comment:
I think it is now better to call numericJobId to something like jobCounter which is what it is. If no custom job id is provided, then this jobCounter will be assigned as the jobId giving it the proper semantics.